OpenRA.Mods.Common/Widgets/ViewportControllerWidget.cs

```diff
@@ -252,11 +252,17 @@ public void UpdateMouseover()
 			if (frozen != null)
 			{
 				var actor = frozen.Actor;
-				if (actor != null && actor.TraitsImplementing<IVisibilityModifier>().All(t => t.IsVisible(actor, world.RenderPlayer)))
+
+				// HACK: This leaks the cloak state through the fog (cloaked buildings will not show tooltips)
```

Comment:
Not even a `HACK`. More like a `BUG`?

Reply:
It is a hack that then causes a bug.

Reply:
IDEs have keyword detection for HACK, but not for BUG, so i'd prefer to keep this as it is.